### PR TITLE
Make mx_military less of a free lunch.

### PR DESCRIPTION
**Pull request by Inglonias**
#### Summary
Balance "Rebalance mx_military map extra to less of a free lunch"

#### Purpose of change
Partially addresses #53245

#### Describe the solution
* New extra-specific variants of the zombie soldier and feral soldier have been created for the map extra. These variants are far more likely to drop rifles and shotguns than their counterparts seen elsewhere.
* The new feral soldier variant has a handgun.
* All weapons dropped on the ground are now gone.

#### Describe alternatives you've considered
Letting #55908 go ahead instead of this. I'm going to wait for that PR to be resolved one way or the other before I undraft this.

#### Testing
I didn't test this yet. That's partly why it's marked as a draft

#### Additional context
This is going to conflict with the changes @KurzedMetal is proposing, but I figured this was the best place to gather my thoughts and seek feedback. I am trying my best to not steal their thunder here.

### CODE REVIEW

**Comment by anothersimulacrum** [comment]
As mentioned on the other PR, I don't like the duplicating of monsters here.

**Comment by Inglonias** [comment]
If there's a better way to handle this than duplicating monster fields, I would love to hear it. I don't suppose I can use `copy-from`, can I?

**Comment by Inglonias** [comment]
Turns out that I can use copy-from. How about that?

File: data/json/monsters/feral_humans.json [hunk 1]
```diff
@@ -690,5 +690,29 @@
...
+    "looks_like": "mon_feral_soldier",
+    "copy-from": "mon_feral_soldier",
+    "description": "Wearing a military uniform with an armored vest, this person seems to be a highly trained military professional, now turned feral.  Tense, dirty hands grip a handgun and combat knife as their bloodshot eyes scan for threats.",
+    "starting_ammo": { "9mm": 3 },
+    "special_attacks": [
+      {
+        "type": "gun",
+        "cooldown": 10,
+        "move_cost": 50,
+        "//": "(movecost is 50 to show that the feral prepper doesn't take time to aim, and just shoots barbaric-ly)",
```
Comment (Maleclypse) [inline data/json/monsters/feral_humans.json:705]:
```suggestion
        "//": "(movecost is 50 to show that the feral solider doesn't take time to aim, and just shoots barbaric-ly)",
```
I'm super guilty of leaving comments in my copy-paste but it probably should be slightly corrected at least.

File: data/json/monsters/feral_humans.json [hunk 2]
```diff
@@ -690,5 +690,29 @@
...
+        "gun_type": "m9",
+        "ammo_type": "9mm",
+        "fake_skills": [ [ "gun", 2 ], [ "pistol", 2 ] ],
+        "fake_dex": 8,
+        "fake_per": 9,
+        "ranges": [ [ 0, 14, "DEFAULT" ], [ 10, 35, "DEFAULT" ] ],
+        "require_targeting_player": false,
+        "description": "The feral prepper haphazardly fires their firearm!"
```
Comment (Maleclypse) [inline data/json/monsters/feral_humans.json:713]:
```suggestion
        "description": "The feral soldier haphazardly fires their firearm!"
```

File: src/map_extras.cpp [hunk 3]
```diff
@@ -581,21 +581,11 @@ static bool mx_military( map &m, const tripoint & )
...
-                        group = Item_spawn_data_military_standard_sniper_rifles;
-                    } else if( one_in( 19 ) ) {
-                        group = Item_spawn_data_military_standard_shotguns;
-                    } else if( one_in( 9 ) ) {
-                        group = Item_spawn_data_military_standard_lmgs;
-                    } else {
-                        group = Item_spawn_data_military_standard_assault_rifles;
-                    }
-                    m.place_items( group, 100, *p, *p, true, calendar::start_of_cataclysm );
+                if ( one_in( 3 ) ) {
```
Comment (Maleclypse) [inline src/map_extras.cpp:584]: Ick, I didn't realize this extra was still hardcoded.

Comment (Inglonias) [reply src/map_extras.cpp:584]: I don't know enough about things to un-hardcode this. If you want to do that, be my guest.

**Comment by ymber** [comment]
Why can a feral soldier fire a pistol but not a rifle? Why can it only fire 3 rounds from its pistol? Why are there so many soldiers with pistols when they're only issued to officers? There's a reason they only spawn on 1/8 soldier zombies. Why are the soldiers at this particular map extra more likely to have had rifles on slings than soldiers elsewhere?

Feral soldiers that can still use some of their gear are a good concept but this changeset hasn't been properly thought through.

**Comment by Inglonias** [comment]
> Why can a feral soldier fire a pistol but not a rifle?
I felt giving them rifles was a little much. If you want a lore reason, maybe they're too dumb from brain rot to figure out how to use a two-handed weapon. Swapping to a rifle makes this excessively powerful in my eyes. That said, I am absolutely open to suggestions here.

> Why can it only fire 3 rounds from its pistol?
I copied this from feral prepper. I'm open to changing this.

> Why are there so many soldiers with pistols when they're only issued to officers? There's a reason they only spawn on 1/8 soldier zombies.
I'm not in the military and didn't know that was how it worked.

> Why are the soldiers at this particular map extra more likely to have had rifles on slings than soldiers elsewhere?
Copy/paste. I'm open to changing this.

**Comment by ymber** [comment]
The weapons they're using need to make sense in setting. A feral using a rifle seems reasonable but its accuracy would be shockingly bad. It wouldn't be a problem of getting killed from the edge of the reality bubble when there's no way the monster is hitting anything past 10 yards.

That also conveniently solves the problem of them all having pistols. There are other equipment problems to think about though. Do we have the infrastructure code it would need for a feral to drop the weapon it was using? It would be a problem for a shotgun wielding feral to drop an M4 on death.

I wrote some infrastructure code back in the day to let turrets spawn with a random amount of ammo in some specified range. You should be able to use that to give these a random amount of rounds up to a magazine's worth.

I can't see any good justification for special variants of the zombie soldier just for this map extra. If you've got armed ferals around you've already solved your easy loot handout problem anyway.

**Pull request closed without merging**